File: src/omnia/data/structure/rx/WritableObservableUndirectedGraph.java
package omnia.data.structure.rx;

import omnia.data.structure.mutable.MutableUndirectedGraph;

public interface WritableObservableUndirectedGraph<E>
    extends MutableUndirectedGraph<E>,
    ObservableUndirectedGraph<E>,
    WritableObservableGraph<E> {

  @Override
  ObservableUndirectedGraph<E> toReadOnly();
}

File: omnia/src/commonMain/omnia/data/structure/observable/writable/WritableObservableUndirectedGraph.kt
package omnia.data.structure.observable.writable

import omnia.data.structure.mutable.MutableUndirectedGraph
import omnia.data.structure.observable.ObservableUndirectedGraph

interface WritableObservableUndirectedGraph<E : Any> :
    MutableUndirectedGraph<E>, ObservableUndirectedGraph<E>, WritableObservableGraph<E> {

  override fun toReadOnly(): ObservableUndirectedGraph<E>
}